rust_snuba/src/processors/functions.rs

```diff
@@ -13,38 +13,38 @@ pub fn process_message(
     _metadata: KafkaMessageMetadata,
 ) -> anyhow::Result<InsertBatch> {
     let payload_bytes = payload.payload().context("Expected payload")?;
-    let msg: FromFunctionsMessage = serde_json::from_slice(payload_bytes)?;
+    let msg: InputMessage = serde_json::from_slice(payload_bytes)?;
 
     let timestamp = match msg.timestamp {
         Some(timestamp) => timestamp,
         _ => SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs(),
     };
     let device_classification = msg.device_class.unwrap_or_default();
 
-    let functions = msg.functions.into_iter().map(|from| {
+    let functions = msg.functions.iter().map(|from| {
         Function {
             profile_id: msg.profile_id,
             project_id: msg.project_id,
             // Profile metadata
-            browser_name: msg.browser_name.clone(),
```

Comment:
i'm curious why this clone was needed if we already used `into_iter` -- should we not have managed to just _move_ the `String` into the new struct?

Reply:
You could have more than one `Function` per `msg`, thats why.